b2sdk/api.py

```diff
@@ -587,6 +587,18 @@ def get_file_info(self, file_id: str) -> FileVersion:
             self.session.get_file_info_by_id(file_id)
         )
 
+    def get_file_info_by_name(self, bucket_name: str, file_name: str) -> FileVersion:
+        """
+        Gets info about file version. Similar to `get-file-info` but 
+        takes the bucket name and file name instead of file id.
+
+        :param str file_id: the id of the file whose info will be retrieved.
+        """
+        return self.file_version_factory.from_api_response(
+            self.session.get_file_info_by_name(bucket_name, file_name),
+            force_action='folder',
```

Comment:
The guy in initial PR you mentioned says that `b2-download-file-by-name` will return 404 for hidden files, so most of the time `action` will be `upload`, but it could also be `start` as far as I understand. Probably there should be a test to start large file upload and try `HEAD b2-download-file-by-name`. If it returns 404 as well then I think it's safe to force `upload` action.
Dunno what is `folder` and how to deal with it.

Reply:
ok, thanks! i actually missed the comment about hidden files in the original issue 🤦

Reply:
@agoncharov-reef looks like `Bucket.get_file_info_by_name()` works correctly for hidden files as well